src/Symfony/Component/Notifier/Bridge/Mastodon/MastodonOptions.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\Mastodon;
+
+use Symfony\Component\Notifier\Message\MessageOptionsInterface;
+
+final class MastodonOptions implements MessageOptionsInterface
+{
+    public function __construct(
+        private array $options = [],
+    ) {
+    }
+
+    public function toArray(): array
+    {
+        return $this->options;
+    }
+
+    public function getRecipientId(): ?string
+    {
+        return null;
+    }
+
+    public function poll(array $options, int $expiresIn): self
+    {
+        $this->options['poll'] = [
+            'options' => $options,
+            'expires_in' => $expiresIn,
+        ];
+
+        return $this;
+    }
+
+    public function media(array $attachmentIds): self
```

Comment:
We need to provide uploading capabilities: fetching the media ids should not be needed.
See twitter PR for what I mean (but of course the way the images are uploaded need to fit the mastodon API)

Reply:
Do you think providing 2 ways to attach media should be ok? 
1. Provide a way to attach already uploaded media (if attachements already exist for example)
2. Provide another way to upload File directly with this transport

Reply:
ppl that need 1. can use the new request method
I think 2. is the thing we need 99% of the time

Reply:
I implemented that but without using streaming responses as you did in Twitter PR, is it ok for you?